product/src/main/java/com/yas/product/service/ProductService.java

```diff
@@ -90,7 +90,7 @@ public ProductListGetVm getProductsWithFilter(int pageNo, int pageSize, String p
 
     public ProductDetailVm getProduct(String slug) {
         Product product = productRepository
-                .findBySlug(slug)
+                .findBySlugAndIsActiveTrue(slug)
```

Comment:
Thank Lam. Good job. I have a small question in this line. Because we want better SEO, we don't need to check isVisiableIndividuallyTrue, right?